### PR TITLE
PB-35568 + multiple files per attachment

### PR DESCRIPTION
**Pull request by michalmaciejewski91**
@SeungChoi , could you take a look please?

### CODE REVIEW

File: sdk/SDK.Examples/src/AttachmentRequirementExample.cs [hunk 1]
```diff
@@ -119,8 +122,11 @@ override public void Execute()
             eslClient.UploadAttachment(packageId, signer1Att1.Id, ATTACHMENT_FILE_NAME1, attachment1ForSigner1FileContent, SIGNER1_ID);
             eslClient.UploadAttachment(packageId, signer2Att1.Id, ATTACHMENT_FILE_NAME2, 
                                        new StreamDocumentSource(attachmentInputStream2).Content(), SIGNER2_ID);
-            eslClient.UploadAttachment(PackageId, signer2Att2.Id, ATTACHMENT_FILE_NAME3, 
-                                       new StreamDocumentSource(attachmentInputStream3).Content(), SIGNER2_ID);
+
+            Dictionary<string, byte []> signer2att2files = new Dictionary<string, byte []> ();
```
Comment (SeungChoi) [inline sdk/SDK.Examples/src/AttachmentRequirementExample.cs]: As with the Java SDK, use the interface for a method parameter. You can use the IDictionary here.

File: sdk/Silanis.ESL.SDK/src/EslClient.cs [hunk 2]
```diff
@@ -563,14 +563,22 @@ public Document UploadDocument(string fileName, byte[] fileContent, Document doc
         }
 
         public void UploadAttachment(PackageId packageId, string attachmentId, string filename, byte[] fileBytes, string signerId) {
+            Dictionary<string, byte []> files = new Dictionary<string, byte []> ();
+            files.Add (filename, fileBytes);
+
+            UploadAttachment (packageId, attachmentId, files, signerId);
+        }
+
+        public void UploadAttachment (PackageId packageId, string attachmentId, Dictionary<string, byte[]> files, string signerId)
```
Comment (SeungChoi) [inline sdk/Silanis.ESL.SDK/src/EslClient.cs]: As with the Java SDK, use the interface for a method parameter. You can use the IDictionary here.

File: sdk/Silanis.ESL.SDK/src/Services/AttachmentRequirementService.cs [hunk 3]
```diff
@@ -102,7 +103,14 @@ public DownloadedFile DownloadAllAttachmentFilesForSignerInPackage(DocumentPacka
 
         public void UploadAttachment(PackageId packageId, string attachmentId, string fileName, byte[] fileBytes, string signerSessionId)
         {
-            apiClient.UploadAttachment(packageId, attachmentId, fileName, fileBytes, signerSessionId);
+            Dictionary<string, byte []> files = new Dictionary<string, byte []> ();
```
Comment (SeungChoi) [inline sdk/Silanis.ESL.SDK/src/Services/AttachmentRequirementService.cs]: As with the Java SDK, use the interface for a method parameter. You can use the IDictionary here.

File: sdk/Silanis.ESL.SDK/src/Services/AttachmentRequirementService.cs [hunk 4]
```diff
@@ -102,7 +103,14 @@ public DownloadedFile DownloadAllAttachmentFilesForSignerInPackage(DocumentPacka
...
+            files.Add (fileName, fileBytes);
+            UploadAttachment(packageId, attachmentId, files, signerSessionId);
+        }
+
+        public void UploadAttachment (PackageId packageId, string attachmentId, Dictionary<string, byte []> files, string signerSessionId)
```
Comment (SeungChoi) [inline sdk/Silanis.ESL.SDK/src/Services/AttachmentRequirementService.cs]: As with the Java SDK, use the interface for a method parameter. You can use the IDictionary here.

File: sdk/Silanis.ESL.SDK/src/Services/Internal/AttachmentRequirementApiClient.cs [hunk 5]
```diff
@@ -165,24 +166,35 @@ private DownloadedFile DownloadAllAttachmentsForSignerInPackage(string packageId
             }
         }
 
-        public void UploadAttachment(PackageId packageId, string attachmentId, string fileName, byte[] fileBytes, string signerSessionId)
+        public void UploadAttachment (PackageId packageId, string attachmentId, string fileName, byte [] fileBytes, string signerSessionId)
+        {
+            Dictionary<string, byte []> files = new Dictionary<string, byte []> ();
```
Comment (SeungChoi) [inline sdk/Silanis.ESL.SDK/src/Services/Internal/AttachmentRequirementApiClient.cs]: As with the Java SDK, use the interface for a method parameter. You can use the IDictionary here.

File: sdk/Silanis.ESL.SDK/src/Services/Internal/AttachmentRequirementApiClient.cs [hunk 6]
```diff
@@ -165,24 +166,35 @@ private DownloadedFile DownloadAllAttachmentsForSignerInPackage(string packageId
...
+            files.Add (fileName, fileBytes);
+            UploadAttachment (packageId, attachmentId, files, signerSessionId);
+        }
+
+        public void UploadAttachment(PackageId packageId, string attachmentId, Dictionary<string, byte[]> files, string signerSessionId)
```
Comment (SeungChoi) [inline sdk/Silanis.ESL.SDK/src/Services/Internal/AttachmentRequirementApiClient.cs]: As with the Java SDK, use the interface for a method parameter. You can use the IDictionary here.

File: sdk/SDK.Examples/src/AttachmentRequirementExample.cs (hunk 1 above)
Comment (michalmaciejewski91) [reply sdk/SDK.Examples/src/AttachmentRequirementExample.cs]: updated as requested

File: sdk/Silanis.ESL.SDK/src/EslClient.cs (hunk 2 above)
Comment (michalmaciejewski91) [reply sdk/Silanis.ESL.SDK/src/EslClient.cs]: updated as requested

File: sdk/Silanis.ESL.SDK/src/Services/AttachmentRequirementService.cs (hunk 3 above)
Comment (michalmaciejewski91) [reply sdk/Silanis.ESL.SDK/src/Services/AttachmentRequirementService.cs]: updated as requested

File: sdk/Silanis.ESL.SDK/src/Services/AttachmentRequirementService.cs (hunk 4 above)
Comment (michalmaciejewski91) [reply sdk/Silanis.ESL.SDK/src/Services/AttachmentRequirementService.cs]: updated as requested

File: sdk/Silanis.ESL.SDK/src/Services/Internal/AttachmentRequirementApiClient.cs (hunk 5 above)
Comment (michalmaciejewski91) [reply sdk/Silanis.ESL.SDK/src/Services/Internal/AttachmentRequirementApiClient.cs]: updated as requested

File: sdk/Silanis.ESL.SDK/src/Services/Internal/AttachmentRequirementApiClient.cs (hunk 6 above)
Comment (michalmaciejewski91) [reply sdk/Silanis.ESL.SDK/src/Services/Internal/AttachmentRequirementApiClient.cs]: updated as requested

**Pull request merged**